### PR TITLE
Add Pull Request template

### LINKED ISSUE #363: Add a Pull Request template
**Is your feature request related to a problem? Please describe.**
I would like to have a Pull Request template to make it easier to contribute to this project.

### PR DESCRIPTION
**Pull request by Benardi**
# Add Pull Request Template

## Description
Add a Pull Request template to root directory to facilitate future contributions.

Fixes #363

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- [x] Forked repository reproduced existing CI tests

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by Benardi** [comment]
Thanks, mate.

**Comment by tallalj** [comment]
@Benardi kindly modify the readme.md and link this in Start Contributing section.

File: pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+- [ ] Test B
+
+## Checklist:
+
+- [ ] My code follows the style guidelines of this project
+- [ ] I have performed a self-review of my own code
+- [ ] I have commented my code, particularly in hard-to-understand areas
+- [ ] I have made corresponding changes to the documentation
+- [ ] My changes generate no new warnings
+- [ ] I have added tests that prove my fix is effective or that my feature works
```
Comment (tallalj) [inline pull_request_template.md]: Just change to `- [] New Unit tests added`

File: pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+- [ ] New and existing unit tests pass locally with my changes
```
Comment (tallalj) [inline pull_request_template.md]: Change to `Unit tests pass locally with my changes`

File: pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+- [ ] Any dependent changes have been merged and published in downstream modules
```
Comment (tallalj) [inline pull_request_template.md]: can you kindly explain this?

File: pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+- [ ] Test A
+- [ ] Test B
+
+## Checklist:
+
+- [ ] My code follows the style guidelines of this project
+- [ ] I have performed a self-review of my own code
+- [ ] I have commented my code, particularly in hard-to-understand areas
+- [ ] I have made corresponding changes to the documentation
+- [ ] My changes generate no new warnings
```
Comment (tallalj) [inline pull_request_template.md]: We don't need these lines. Instead just write:
`- [] I did follow the [contribution guidelines](https://github.com/OpenMined/PySyft/blob/master/CONTRIBUTING.md)`

File: pull_request_template.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+## Description
+
+Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
+
+Fixes # (issue)
+
+## Type of change
+
+Please delete options that are not relevant.
```
Comment (tallalj) [inline pull_request_template.md]: Kindly check options that are relevant

File: pull_request_template.md (hunk 3 above)
Comment (Benardi) [reply pull_request_template.md]: What line?

File: pull_request_template.md (hunk 5 above)
Comment (Benardi) [reply pull_request_template.md]: I would argue in favor of something in the line of:

Please mark options that are relevant.

**Comment by IonesioJunior** [comment]
Hello @tallalj, do you have another request for this PR?

**Comment by tallalj** [comment]
looks awesome to me  @IonesioJunior  :)

File: pull_request_template.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+## How Has This Been Tested?
+
+Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
+
+- [ ] Test A
+- [ ] Test B
+
+## Checklist:
+
+- [] I did follow the [contribution guidelines](https://github.com/OpenMined/PySyft/blob/master/CONTRIBUTING.md)
```
Comment (tallalj) [inline pull_request_template.md]: typo in `[ ]`. Missing a space

**Comment by IonesioJunior** [comment]
Awesome! Thanks, @tallalj, @hericlesme, and @Benardi for this amazing contribution! : rocket:

**Pull request merged**